pkg/apis/config/feature_flags.go

```diff
@@ -202,12 +227,12 @@ func NewFeatureFlagsFromMap(cfgMap map[string]string) (*FeatureFlags, error) {
 	// defeat the purpose of having a single shared gate for all alpha features.
 	if tc.EnableAPIFields == AlphaAPIFields {
 		tc.EnableTektonOCIBundles = true
-		tc.EnableSpire = true
+		tc.EnforceNonfalsifiability = EnforceNonfalsifiabilityWithSpire
```

Comment:
does this mean that if alpha fields are on, enforcing with spire will be turned on also? i think we'd want to configure spire enforcement on _only_ explicitly - i.e. its only configurable if alpha fields are turned on, but by default it's off

Reply:
+1 agree that Spire should be explicitly turned on as it has an external dependency on Spire running in the environment. The reason I did this originally was because of the following comment:
```
	// Note: the user cannot enable "alpha" while disabling bundles or custom tasks - that would
	// defeat the purpose of having a single shared gate for all alpha features.
```
"single shared gate for all alpha features" can this be removed?

Reply:
Though the comment "single shared gate for all alpha features" remains, the code is modified so that SPIRE is enabled only when alpha is enabled.
If SPIRE flag is set in the stable/beta modes, SPIRE will still not be enabled.

Reply:
To be clear setting the alpha flag should enable the ability to _use_ all alpha features but not necessarily configure them and turn them on. I think there are two things getting mixed together here:
1. Features have stability levels associated with them (alpha/beta/stable) and we are letting operators opt into making these features available via the EnableAPIFields flag
2. Some features are optional (regardless of their stability level), including using SPIRE

SPIRE support will initially be alpha, so the EnableAPIFields should be required to be configured at alpha in order for the SPIRE functionality to be available. However there should still be more configuration required (EnforceNonfalsifiability = SPIRE) in order to turn it on

> Changing the contract that setting alpha flag will enable all alpha features

Hopefully this shouldn't be a change in the contract and should match the description of the alpha feature flag in [TEP-0033](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-feature-gates.md) (or maybe TEP-0033 needs to be updated with more details about a case like this where a feature is alpha but also needs to be opted into? on the other hand it seems like the trend is toward per-feature flags in the future based on some recent discussions)

> Enable SPIRE only if alpha is set and EnforceNonfalsifiability is set to spire.

+1

> Even if the EnforceNonfalsifiability flag is set to spire, if alpha is not enabled then do not enable spire.

Yes this makes sense to me - this could even be treated as an error

Reply:
Yes that makes sense.

Reply:
As mentioned in the previous comment, https://github.com/tektoncd/pipeline/pull/5902#discussion_r1072875138, this has been fixed as suggested, i.e. SPIRE can only be enabled if alpha is turned on and SPIRE will not be enabled even if it is set but alpha is not turned on.